### PR TITLE
[tool] Switch PR description overrides over to labels

### PR DESCRIPTION
**Pull request by stuartmorgan**
For everything but Dependabot auto-allow checks, switch from using the PR description to labels, since it turns out that we don't reliably have access to the PR description (but instead the PR description in some cases and the last incremental commit messages in others).

Fixes https://github.com/flutter/flutter/issues/107485

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/main/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by stuartmorgan** [comment]
I verified (above) that re-running `publishable` after changing labels works 🎉

**Comment by ditman** [comment]
This is a great change!

File: .cirrus.yml [hunk 1]
```diff
@@ -88,17 +88,17 @@ task:
       env:
         CHANGE_DESC: "$TMPDIR/change-description.txt"
       version_check_script:
-        # For pre-submit, pass the PR description to the script to allow for
-        # version check overrides.
+        # For pre-submit, pass the PR label, as well as the PR description or
+        # incremenal commit message (for Dependabot checks), to the script to
```
Comment (ditman) [inline .cirrus.yml:92]:
```suggestion
        # incremental commit message (for Dependabot checks), to the script to
```

File: script/tool/lib/src/version_check_command.dart [hunk 2]
```diff
@@ -123,47 +123,49 @@ class VersionCheckCommand extends PackageLoopingCommand {
...
   static const String _ignorePlatformInterfaceBreaks =
       'ignore-platform-interface-breaks';
 
-  /// The string that must be in [_changeDescriptionFile] to allow a breaking
+  /// The label that must be on a PR to allow a breaking
   /// change to a platform interface.
-  static const String _breakingChangeJustificationMarker =
-      '## Breaking change justification';
+  static const String _breakingChangeOverrideLabel =
+      'override: allow breaking change';
```
Comment (ditman) [inline script/tool/lib/src/version_check_command.dart:158]: Even if people can change the name of the label inadvertently in the Github UI, the result is that the check CANNOT be overridden, so the failure mode is safety, good stuff!

File: .cirrus.yml (hunk 1 above)
Comment (ditman) [inline .cirrus.yml:92]:
```suggestion
        # incremental commit message (for Dependabot checks), to the script to
```

**Review by ditman — APPROVED** [review]
I really like this new approach!

**Pull request merged**